transaction-scenarios/src/scenarios/fungible_resource.rs

```diff
@@ -0,0 +1,435 @@
+pub struct FungibleResourceScenarioState {
+    pub max_divisibility_fungible_resource: Option<ResourceAddress>,
+    pub min_divisibility_fungible_resource: Option<ResourceAddress>,
+    pub vault1: Option<InternalAddress>,
+    pub vault2: Option<InternalAddress>,
+}
+
+impl Default for FungibleResourceScenarioConfig {
+    fn default() -> Self {
+        Self {
+            user_account_1: secp256k1_account_1(),
+            user_account_2: secp256k1_account_2(),
+        }
+    }
+}
+
+pub struct FungibleResourceScenarioCreator;
+
+impl ScenarioCreator for FungibleResourceScenarioCreator {
+    type Config = FungibleResourceScenarioConfig;
+
+    type State = FungibleResourceScenarioState;
+
+    fn create_with_config_and_state(
+        core: ScenarioCore,
+        config: Self::Config,
+        start_state: Self::State,
+    ) -> Box<dyn ScenarioInstance> {
+        let metadata = ScenarioMetadata {
+            logical_name: "fungible_resource",
+        };
+
+        #[allow(unused_variables)]
+        ScenarioBuilder::new(core, metadata, config, start_state)
+            .successful_transaction_with_result_handler(
+                |core, config, state| {
+                    core.next_transaction_with_faucet_lock_fee(
+                        "fungible-max-div-create",
+                        |builder| {
+                            builder
+                                .create_fungible_resource(
+                                    OwnerRole::None,
+                                    false,
+                                    18,
+                                    metadata! {},
+                                    btreemap! {
+                                        Mint => (rule!(allow_all), rule!(deny_all)),
+                                        Burn =>  (rule!(allow_all), rule!(deny_all)),
+                                        UpdateNonFungibleData => (rule!(allow_all), rule!(deny_all)),
+                                        Withdraw => (rule!(allow_all), rule!(deny_all)),
+                                        Deposit => (rule!(allow_all), rule!(deny_all)),
+                                        Recall => (rule!(allow_all), rule!(deny_all)),
+                                        Freeze => (rule!(allow_all), rule!(deny_all)),
+                                    },
+                                    Some(dec!("100000")),
+                                )
+                                .try_deposit_batch_or_abort(config.user_account_1.address)
+                        },
+                        vec![],
+                    )
+                },
+                |core, config, state, result| {
+                    state.max_divisibility_fungible_resource = Some(result.new_resource_addresses()[0]);
+                    state.vault1 = Some(result.new_vault_addresses()[0]);
+                    Ok(())
+                },
+            )
+            .successful_transaction_with_result_handler(
```

Comment:
You can just use `successful_transaction` to avoid the empty result handler, but I'm happy if you want to keep this for symmetry :)